packages/itwin/reports-config-widget/src/test/Reports.test.tsx

```diff
@@ -43,38 +41,30 @@ const reportsFactory = (): ReportCollection => ({
   },
 });
 
-const server = setupServer();
+const mockGetReports = jest.fn();
+const mockDeleteReport = jest.fn();
+
+const mockReportsClient = moq.Mock.ofType<ReportsClient>();
 
 beforeAll(async () => {
   const localization = new EmptyLocalization();
   await ReportsConfigWidget.initialize(localization);
-  server.listen();
-});
 
-afterAll(() => {
-  server.close();
+  mockReportsClient.setup(async (x) => x.getReports(moq.It.isAny(), moq.It.isAny())).returns(mockGetReports);
+  mockReportsClient.setup(async (x) => x.deleteReport(moq.It.isAny(), moq.It.isAny())).returns(mockDeleteReport);
 });
 
-afterEach(() => server.resetHandlers());
+afterEach(() => {
+  mockGetReports.mockReset();
+  mockDeleteReport.mockReset();
+});
 
 describe("Reports View", () => {
   it("call to action button should be clickable with no reports", async () => {
-    server.use(
-      rest.get(
-        `${REPORTS_CONFIG_BASE_URL}/insights/reporting/reports`,
-        async (_req, res, ctx) => {
-          return res(
-            ctx.delay(500),
-            ctx.status(200),
-            ctx.json({ reports: [] })
-          );
-        }
-      )
-    );
-
+    mockGetReports.mockReturnValueOnce([]);
     const onClickAddMock = jest.fn();
 
-    const { user } = render(<Reports onClickAddReport={onClickAddMock} />);
+    const { user } = render(<Reports onClickAddReport={onClickAddMock} />, { reportsClient: mockReportsClient.object });
 
     await waitForElementToBeRemoved(() => screen.getByText(/loading/i));
```

Comment:
I think the most flakyness was caused by these `waitForElementToBeRemoved` by the element text.